### PR TITLE
[BE] Task #210322 Gender data input correcttion

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/210322

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved the OKYC update process by ensuring gender-related information is correctly verified and modified.

### CODE REVIEW

File: src/src/facilitator/facilitator.service.ts [hunk 1]
```diff
@@ -2452,6 +2452,14 @@ export class FacilitatorService {
 		let message = 'Okyc Details already Updated';
 		let status = 200;
 		if (userData?.aadhar_verified === 'yes') {
+			// Check and modify the gender field in the body if it's 'M' or 'F'
+			if (
+				body?.gender &&
+				(body?.gender === 'M' || body?.gender === 'F')
+			) {
+				body.gender = body?.gender === 'M' ? 'male' : 'female';
+			}
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/facilitator/facilitator.service.ts:2461]: convert this to Utility function to transform gender

[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/facilitator/facilitator.service.ts]: Resolved

File: src/src/common/method/method.ts [hunk 2]
```diff
@@ -102,4 +102,10 @@ export class Method {
 			return false;
 		}
 	}
+
+	public async transformGender(gender) {
+		if (gender === 'M') return 'male';
+		if (gender === 'F') return 'female';
+		return gender;
+	}
 }
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/common/method/method.ts:111]: This method change is not related to this PR raised

**Comment by GitVivekHub** [comment]
Incorrect PR branch, should have been 1.7.0 (cohort) instead of 1.5.10 (hotfix)

**Comment by GitVivekHub** [comment]
Incorrect PR branch, should have been 1.7.0 (cohort) instead of 1.5.10 (hotfix)

**Pull request closed without merging**